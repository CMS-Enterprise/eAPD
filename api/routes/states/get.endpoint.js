const {
  getDB,
  getFullPath,
  login,
  request,
  unauthenticatedTest
} = require('../../utils.endpoint');

describe('states endpoint | GET /states', async () => {
  const url = getFullPath('/states');

  const db = getDB();
  beforeAll(() => db.seed.run());
  afterAll(() => db.destroy());

  unauthenticatedTest('get', url);

  it('when authenticated as a user without an associated state', async () => {
    const cookies = await login('all-permissions-no-state', 'password');
    const { response, body } = await request.get(url, {
      jar: cookies,
      json: true
    });

    expect(response.statusCode).toEqual(401);
    expect(body).toMatchSnapshot();
  });

  it('when authenticated as a user with an associated state', async () => {
    const cookies = await login();
    const { response, body } = await request.get(url, {
      jar: cookies,
      json: true
    });

    expect(response.statusCode).toEqual(200);
    expect(body).toMatchSnapshot();
  });
});

describe('states/:id endpoint | GET /states/:id', async () => {
  const url = getFullPath('/states/mn');

  const db = getDB();
  beforeAll(() => db.seed.run());
  afterAll(() => db.destroy());

  unauthenticatedTest('get', url);

  describe('when authenticated as a user with permission', async () => {
    let cookies;
    beforeAll(async () => {
      cookies = await login();
    });

    it('with a non-existant state ID', async () => {
      const { response, body } = await request.get(
        getFullPath('/states/baloney'),
        {
          jar: cookies,
          json: true
        }
      );

      expect(response.statusCode).toEqual(404);
      expect(body).toMatchSnapshot();
    });

    it('with a real state ID', async () => {
      const { response, body } = await request.get(url, {
        jar: cookies,
        json: true
      });

      expect(response.statusCode).toEqual(200);
      expect(body).toMatchSnapshot();
    });
  });
});
