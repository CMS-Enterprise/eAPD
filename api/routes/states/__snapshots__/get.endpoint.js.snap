// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`states endpoint | GET /states when authenticated as a user with an associated state 1`] = `
Object {
  "id": "mn",
  "medicaid_office": Object {
    "address1": "100 Round Sq",
    "city": "Cityville",
    "director": Object {
      "email": "c.fudge@ministry.magic",
      "name": "Cornelius Fudge",
      "phone": "5551234567",
    },
    "state": "Minnesota",
    "zip": "12345",
  },
  "name": "Minnesota",
  "program_benefits": "The program will have benefits",
  "program_vision": "The program vision is 20/20",
  "state_pocs": Array [
    Object {
      "email": "corinne@thatplace",
      "name": "Corinne Johnson",
      "position": "Head of Muggle Studies",
    },
  ],
}
`;

exports[`states endpoint | GET /states when authenticated as a user without an associated state 1`] = `undefined`;

exports[`states/:id endpoint | GET /states/:id when authenticated as a user with permission with a non-existant state ID 1`] = `undefined`;

exports[`states/:id endpoint | GET /states/:id when authenticated as a user with permission with a real state ID 1`] = `
Object {
  "id": "mn",
  "medicaid_office": Object {
    "address1": "100 Round Sq",
    "city": "Cityville",
    "director": Object {
      "email": "c.fudge@ministry.magic",
      "name": "Cornelius Fudge",
      "phone": "5551234567",
    },
    "state": "Minnesota",
    "zip": "12345",
  },
  "name": "Minnesota",
  "program_benefits": "The program will have benefits",
  "program_vision": "The program vision is 20/20",
  "state_pocs": Array [
    Object {
      "email": "corinne@thatplace",
      "name": "Corinne Johnson",
      "position": "Head of Muggle Studies",
    },
  ],
}
`;
